support/objc-support.m

```diff
@@ -146,3 +146,129 @@ MonoDecimal mono_embeddinator_get_system_decimal (NSDecimalNumber* nsdecimalnumb
 
 	return mdecimal;
 }
+
+long long mono_embeddinator_get_nsdatereferencensdate_ticks (mono_embeddinator_context_t* context)
+{
+	// NSDate reference date 00:00:00 UTC on 1 January 2001
+	// https://developer.apple.com/reference/foundation/nsdate
+	int year = 2001, month = 1, day = 1, hour = 0, minute = 0, second = 0;
+	MonoClass* datetimeclass = mono_embeddinator_get_datetime_class ();
+	int utc = E4KDateTimeKind_Utc;
+
+	void* datetimeargs [7];
+	datetimeargs[0] = &year;
+	datetimeargs[1] = &month;
+	datetimeargs[2] = &day;
+	datetimeargs[3] = &hour;
+	datetimeargs[4] = &minute;
+	datetimeargs[5] = &second;
+	datetimeargs[6] = &utc;
+
+	MonoMethod* datetimector = mono_embeddinator_lookup_method ("System.DateTime:.ctor(int,int,int,int,int,int,System.DateTimeKind)", datetimeclass);
+	MonoMethod* datetimeticks = mono_embeddinator_lookup_method ("System.DateTime:get_Ticks()", datetimeclass);
+	MonoObject* datetimeshell = mono_object_new (context->domain, datetimeclass);
+	void* datetime = mono_object_unbox (datetimeshell);
+
+	MonoObject* ex = nil;
+	mono_runtime_invoke (datetimector, datetime, datetimeargs, &ex);
+	if (ex)
+		mono_embeddinator_throw_exception (ex);
+
+	MonoObject* ex2 = nil;
+	MonoObject* boxedticks = mono_runtime_invoke (datetimeticks, datetime, NULL, &ex2);
+	if (ex2)
+		mono_embeddinator_throw_exception (ex);
+
+	long long ticks = *(long long *) mono_object_unbox (boxedticks);
+
+	return ticks;
+}
+
+static long long nsdaterefdateticks = 0;
+
+NSDate* mono_embeddinator_get_nsdate (void* datetime, mono_embeddinator_context_t* context)
```

Comment:
Can you add a strongly defined type for `datetime` here (and everywhere else you're using `void*`) instead of `void*`? That makes errors harder (and more visible in the code review), and it's also easier to debug code (in lldb).